dace/optim/map_tiling_tuner.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
+import copy
+
+import dace
+import math
+
+import itertools
+from dace.sdfg.sdfg import SDFG
+import numpy as np
+
+from dace.sdfg.analysis.cutout import cutout_state
+from dace.optim.cutout_tuner import CutoutTuner
+from dace.transformation.dataflow import MapTiling
+
+from dace.transformation.optimizer import Optimizer
+from dace.transformation.helpers import get_parent_map
+
+class MapTilingTuner(CutoutTuner):
+
+    def __init__(self, sdfg: dace.SDFG) -> None:
+        super().__init__(sdfg=sdfg)
+
+    def space(self, parent_map):
+        # TODO: choices
+        choices = [
+            (8, 8, 8),
+            (16, 16, 16),
+            (32, 32, 32),
+            (64, 64, 64),
+            (128, 128, 128),
+            (256, 256, 256),
+            (4, 8, 16),
+            (8, 16, 32),
+            (16, 32, 64),
+            (32, 64, 128),
+            (64, 128, 256),
+            (64, 8, 8)
+        ]
+
+        return choices
+
+    def optimize(self, measurements: int = 30):
+        dreport = self._sdfg.get_instrumented_data()
+        
+        tuning_report = {}
+        for state, parent_map in self.cutouts():
+            subgraph_nodes = state.scope_subgraph(parent_map).nodes()
+            cutout = cutout_state(state, *subgraph_nodes)
+            cutout.instrument = dace.InstrumentationType.Timer
+
+            arguments = {}
+            for cstate in cutout.nodes():
+                for dnode in cstate.data_nodes():
+                    if cutout.arrays[dnode.data].transient:
+                        continue
+
+                    arguments[dnode.data] = np.copy(dreport.get_first_version(dnode.data))
+
+            results = {}
+            baseline = self.measure(cutout, arguments, measurements)
+            results[None] = baseline
+
+            best_choice = None
+            best_runtime = baseline
+            cutout_json = cutout.to_json()
+            for point in self.space(parent_map):
+                # TODO: Hacky? Faster than deepcopy?
+                tiled_sdfg = SDFG.from_json(cutout_json)
+                # TODO: Inefficient
+                for xform in Optimizer(tiled_sdfg).get_pattern_matches(patterns=[MapTiling], options={"tile_sizes": point}):
```

Comment:
you should probably use `apply_to` on a particular map entry node, as received from cutouts (`parent_map`)?